src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -294,6 +297,22 @@ private FolioExecutionContext prepareContextForTenant(String tenantId, FolioExec
     return new DefaultFolioExecutionContext(folioModuleMetadata, context.getOkapiHeaders());
   }
 
+  public void deleteShadowUsers(UUID userId) {
+    FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+    List<UserTenantEntity> userTenantEntities = userTenantRepository.getByUserIdAndIsPrimaryFalse(userId);
+    if (CollectionUtils.isNotEmpty(userTenantEntities)) {
+      Map<String, List<UserTenantEntity>> listMap = userTenantEntities.stream().collect(Collectors.groupingBy(userTenantEntity -> userTenantEntity.getTenant().getId()));
```

Comment:
It can by simplified, we don't need map here, we have the same userId and different tenant Ids,
so we can write something like this: List<String> tenantIds = userTenantEntities.stream().map(UserTenantEntity::tenantId) 

Reply:
and so we don't need to use batch endpoint now, because deleting single user from 1 tenant now, so can you use simple delete by id

Reply:
fixed